# Analysis of SuperStoreData

In [1]:
import pandas as pd

In [2]:
df= pd.read_csv('SuperStoreOrders.csv')


## Head

In [3]:
df.head(5)

order_id order_date ship_date       ship_mode    customer_name  \
0     AG-2011-2040   1/1/2011  6/1/2011  Standard Class  Toby Braunhardt   
1    IN-2011-47883   1/1/2011  8/1/2011  Standard Class      Joseph Holt   
2     HU-2011-1220   1/1/2011  5/1/2011    Second Class    Annie Thurman   
3  IT-2011-3647632   1/1/2011  5/1/2011    Second Class     Eugene Moren   
4    IN-2011-47883   1/1/2011  8/1/2011  Standard Class      Joseph Holt   

       segment            state    country  market   region  ...  \
0     Consumer      Constantine    Algeria  Africa   Africa  ...   
1     Consumer  New South Wales  Australia    APAC  Oceania  ...   
2     Consumer         Budapest    Hungary    EMEA     EMEA  ...   
3  Home Office        Stockholm     Sweden      EU    North  ...   
4     Consumer  New South Wales  Australia    APAC  Oceania  ...   

          category sub_category                 product_name sales quantity  \
0  Office Supplies      Storage          Tenex Lockers, Blue   408        2   
1  Office Supplies     Supplies     Acme Trimmer, High Speed   120        3   
2  Office Supplies      Storage      Tenex Box, Single Width    66        4   
3  Office Supplies        Paper  Enermax Note Cards, Premium    45        3   
4        Furniture  Furnishings   Eldon Light Bulb, Duo Pack   114        5   

   discount   profit  shipping_cost  order_priority  year  
0       0.0  106.140          35.46          Medium  2011  
1       0.1   36.036           9.72          Medium  2011  
2       0.0   29.640           8.17            High  2011  
3       0.5  -26.055           4.82            High  2011  
4       0.1   37.770           4.70          Medium  2011  

[5 rows x 21 columns]

## Tail

In [4]:
df.tail(4)

order_id  order_date ship_date       ship_mode     customer_name  \
51286    MO-2014-2560  31-12-2014  5/1/2015  Standard Class         Liz Preis   
51287  MX-2014-110527  31-12-2014  2/1/2015    Second Class  Charlotte Melton   
51288  MX-2014-114783  31-12-2014  6/1/2015  Standard Class     Tamara Dahlen   
51289  CA-2014-156720  31-12-2014  4/1/2015  Standard Class     Jill Matthias   

        segment             state        country  market   region  ...  \
51286  Consumer  Souss-Massa-Draâ        Morocco  Africa   Africa  ...   
51287  Consumer           Managua      Nicaragua   LATAM  Central  ...   
51288  Consumer         Chihuahua         Mexico   LATAM    North  ...   
51289  Consumer          Colorado  United States      US     West  ...   

              category sub_category                             product_name  \
51286  Office Supplies      Binders  Wilson Jones Hole Reinforcements, Clear   
51287  Office Supplies       Labels   Hon Color Coded Labels, 5000 Label Set   
51288  Office Supplies       Labels   Hon Legal Exhibit Labels, Alphabetical   
51289  Office Supplies    Fasteners                      Bagged Rubber Bands   

      sales quantity  discount   profit  shipping_cost  order_priority  year  
51286     4        1       0.0   0.4200           0.49          Medium  2014  
51287    26        3       0.0  12.3600           0.35          Medium  2014  
51288     7        1       0.0   0.5600           0.20          Medium  2014  
51289     3        3       0.2  -0.6048           0.17          Medium  2014  

[4 rows x 21 columns]

### length of Dataset ?

In [5]:
df.shape

(51290, 21)

### Numberof the Columns?

In [6]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'customer_name',
       'segment', 'state', 'country', 'market', 'region', 'product_id',
       'category', 'sub_category', 'product_name', 'sales', 'quantity',
       'discount', 'profit', 'shipping_cost', 'order_priority', 'year'],
      dtype='object')

# Cleaning the Dataset

### Droping the unnecessary columns

In [7]:
df=df.drop('customer_name', axis=1)
df.columns

Index(['order_id', 'order_date', 'ship_date', 'ship_mode', 'segment', 'state',
       'country', 'market', 'region', 'product_id', 'category', 'sub_category',
       'product_name', 'sales', 'quantity', 'discount', 'profit',
       'shipping_cost', 'order_priority', 'year'],
      dtype='object')

## OR

In [8]:
df=df.drop(['ship_mode','order_priority'], axis=1)
df.columns

Index(['order_id', 'order_date', 'ship_date', 'segment', 'state', 'country',
       'market', 'region', 'product_id', 'category', 'sub_category',
       'product_name', 'sales', 'quantity', 'discount', 'profit',
       'shipping_cost', 'year'],
      dtype='object')

In [9]:
df.shape

(51290, 18)

In [10]:
# Only Selecting some columns
df[['order_date','order_id','sales','state','country','market']]

order_date         order_id sales             state        country  \
0        1/1/2011     AG-2011-2040   408       Constantine        Algeria   
1        1/1/2011    IN-2011-47883   120   New South Wales      Australia   
2        1/1/2011     HU-2011-1220    66          Budapest        Hungary   
3        1/1/2011  IT-2011-3647632    45         Stockholm         Sweden   
4        1/1/2011    IN-2011-47883   114   New South Wales      Australia   
...           ...              ...   ...               ...            ...   
51285  31-12-2014   CA-2014-115427    14        California  United States   
51286  31-12-2014     MO-2014-2560     4  Souss-Massa-Draâ        Morocco   
51287  31-12-2014   MX-2014-110527    26           Managua      Nicaragua   
51288  31-12-2014   MX-2014-114783     7         Chihuahua         Mexico   
51289  31-12-2014   CA-2014-156720     3          Colorado  United States   

       market  
0      Africa  
1        APAC  
2        EMEA  
3          EU  
4        APAC  
...       ...  
51285      US  
51286  Africa  
51287   LATAM  
51288   LATAM  
51289      US  

[51290 rows x 6 columns]

### Datatype of Columns?

In [11]:
df.dtypes

order_id          object
order_date        object
ship_date         object
segment           object
state             object
country           object
market            object
region            object
product_id        object
category          object
sub_category      object
product_name      object
sales             object
quantity           int64
discount         float64
profit           float64
shipping_cost    float64
year               int64
dtype: object

### changing the datatype of required columns.

In [12]:
df['order_date']=pd.to_datetime(df['order_date'])
df['ship_date']=pd.to_datetime(df['ship_date'])

C:\Users\Prajwal Singh\AppData\Roaming\Python\Python310\site-packages\pandas\core\tools\datetimes.py:1047: UserWarning: Parsing '13-01-2011' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  cache_array = _maybe_cache(arg, format, cache, convert_listlike)
C:\Users\Prajwal Singh\AppData\Roaming\Python\Python310\site-packages\pandas\core\tools\datetimes.py:1047: UserWarning: Parsing '14-01-2011' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  cache_array = _maybe_cache(arg, format, cache, convert_listlike)
C:\Users\Prajwal Singh\AppData\Roaming\Python\Python310\site-packages\pandas\core\tools\datetimes.py:1047: UserWarning: Parsing '15-01-2011' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  cache_array = _maybe_cache(arg, format, cache, convert_listlike)
C:\Users\Prajwal Singh\AppData\Roaming\Python\Python310\site-packages\pandas\co

In [13]:
df.dtypes

order_id                 object
order_date       datetime64[ns]
ship_date        datetime64[ns]
segment                  object
state                    object
country                  object
market                   object
region                   object
product_id               object
category                 object
sub_category             object
product_name             object
sales                    object
quantity                  int64
discount                float64
profit                  float64
shipping_cost           float64
year                      int64
dtype: object

### Adding column?

In [14]:
df['delivery_days']=df['ship_date']-df['order_date']


In [15]:
df['delivery_days']=pd.to_datetime(df['delivery_days'])
df.dtypes

TypeError: dtype timedelta64[ns] cannot be converted to datetime64[ns]

In [16]:
df.columns

Index(['order_id', 'order_date', 'ship_date', 'segment', 'state', 'country',
       'market', 'region', 'product_id', 'category', 'sub_category',
       'product_name', 'sales', 'quantity', 'discount', 'profit',
       'shipping_cost', 'year', 'delivery_days'],
      dtype='object')

In [17]:
df.head(3)

order_id order_date  ship_date   segment            state    country  \
0   AG-2011-2040 2011-01-01 2011-06-01  Consumer      Constantine    Algeria   
1  IN-2011-47883 2011-01-01 2011-08-01  Consumer  New South Wales  Australia   
2   HU-2011-1220 2011-01-01 2011-05-01  Consumer         Budapest    Hungary   

   market   region        product_id         category sub_category  \
0  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   
2    EMEA     EMEA  OFF-TEN-10001585  Office Supplies      Storage   

               product_name sales  quantity  discount   profit  shipping_cost  \
0       Tenex Lockers, Blue   408         2       0.0  106.140          35.46   
1  Acme Trimmer, High Speed   120         3       0.1   36.036           9.72   
2   Tenex Box, Single Width    66         4       0.0   29.640           8.17   

   year delivery_days  
0  2011      151 days  
1  2011      212 days  
2  2011      120 days

In [18]:
df[['order_date','ship_date','delivery_days']]

order_date  ship_date delivery_days
0     2011-01-01 2011-06-01      151 days
1     2011-01-01 2011-08-01      212 days
2     2011-01-01 2011-05-01      120 days
3     2011-01-01 2011-05-01      120 days
4     2011-01-01 2011-08-01      212 days
...          ...        ...           ...
51285 2014-12-31 2015-04-01       91 days
51286 2014-12-31 2015-05-01      121 days
51287 2014-12-31 2015-02-01       32 days
51288 2014-12-31 2015-06-01      152 days
51289 2014-12-31 2015-04-01       91 days

[51290 rows x 3 columns]

### finding nan values in selected columns ?

In [19]:
df['state'].isna().sum()

0

### Total NAN values in dataframe

In [20]:
df[:].isna().sum()

order_id         0
order_date       0
ship_date        0
segment          0
state            0
country          0
market           0
region           0
product_id       0
category         0
sub_category     0
product_name     0
sales            0
quantity         0
discount         0
profit           0
shipping_cost    0
year             0
delivery_days    0
dtype: int64

In [21]:
df.head(3)

order_id order_date  ship_date   segment            state    country  \
0   AG-2011-2040 2011-01-01 2011-06-01  Consumer      Constantine    Algeria   
1  IN-2011-47883 2011-01-01 2011-08-01  Consumer  New South Wales  Australia   
2   HU-2011-1220 2011-01-01 2011-05-01  Consumer         Budapest    Hungary   

   market   region        product_id         category sub_category  \
0  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   
2    EMEA     EMEA  OFF-TEN-10001585  Office Supplies      Storage   

               product_name sales  quantity  discount   profit  shipping_cost  \
0       Tenex Lockers, Blue   408         2       0.0  106.140          35.46   
1  Acme Trimmer, High Speed   120         3       0.1   36.036           9.72   
2   Tenex Box, Single Width    66         4       0.0   29.640           8.17   

   year delivery_days  
0  2011      151 days  
1  2011      212 days  
2  2011      120 days

In [22]:
df.describe

<bound method NDFrame.describe of               order_id order_date  ship_date      segment             state  \
0         AG-2011-2040 2011-01-01 2011-06-01     Consumer       Constantine   
1        IN-2011-47883 2011-01-01 2011-08-01     Consumer   New South Wales   
2         HU-2011-1220 2011-01-01 2011-05-01     Consumer          Budapest   
3      IT-2011-3647632 2011-01-01 2011-05-01  Home Office         Stockholm   
4        IN-2011-47883 2011-01-01 2011-08-01     Consumer   New South Wales   
...                ...        ...        ...          ...               ...   
51285   CA-2014-115427 2014-12-31 2015-04-01    Corporate        California   
51286     MO-2014-2560 2014-12-31 2015-05-01     Consumer  Souss-Massa-Draâ   
51287   MX-2014-110527 2014-12-31 2015-02-01     Consumer           Managua   
51288   MX-2014-114783 2014-12-31 2015-06-01     Consumer         Chihuahua   
51289   CA-2014-156720 2014-12-31 2015-04-01     Consumer          Colorado   

             coun

### How many duplicate records are there?

In [23]:
df[df.duplicated()]

Empty DataFrame
Columns: [order_id, order_date, ship_date, segment, state, country, market, region, product_id, category, sub_category, product_name, sales, quantity, discount, profit, shipping_cost, year, delivery_days]
Index: []

### Null Values?

In [24]:
df.isnull().sum()

order_id         0
order_date       0
ship_date        0
segment          0
state            0
country          0
market           0
region           0
product_id       0
category         0
sub_category     0
product_name     0
sales            0
quantity         0
discount         0
profit           0
shipping_cost    0
year             0
delivery_days    0
dtype: int64

In [25]:
df.head(2)

order_id order_date  ship_date   segment            state    country  \
0   AG-2011-2040 2011-01-01 2011-06-01  Consumer      Constantine    Algeria   
1  IN-2011-47883 2011-01-01 2011-08-01  Consumer  New South Wales  Australia   

   market   region        product_id         category sub_category  \
0  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   

               product_name sales  quantity  discount   profit  shipping_cost  \
0       Tenex Lockers, Blue   408         2       0.0  106.140          35.46   
1  Acme Trimmer, High Speed   120         3       0.1   36.036           9.72   

   year delivery_days  
0  2011      151 days  
1  2011      212 days

### Number of records where country in 'Algeria'

In [26]:
df[df['country'].isin(['Algeria'])]['country'].count()


196

### Avarage discount given by Superstore ?

In [27]:
df['discount'].mean().round(5)

0.14291

### Avarge delivery days

In [28]:
df['delivery_days'].mean()

Timedelta('5 days 01:53:49.120686293')

## (Timedelta('5 days 01:53:49.120686293')) - days required to get better Sales

### How many years sales is ?

In [29]:
df['year'].unique()

array([2011, 2012, 2013, 2014], dtype=int64)

### sales in 2013?

In [30]:
df[df['year']==2013][['sales','year','sub_category','product_name' ]]

sales  year sub_category                            product_name
19960  1,649  2013      Copiers             Canon Copy Machine, Digital
19961  1,358  2013    Bookcases        Sauder Floating Shelf Set, Metal
19962    729  2013  Accessories         Logitech Memory Card, Erganomic
19963  2,190  2013    Bookcases        Dania Library with Doors, Mobile
19964  1,362  2013    Bookcases  Sauder Floating Shelf Set, Traditional
...      ...   ...          ...                                     ...
33754     10  2013  Furnishings                 Eldon Light Bulb, Black
33755     17  2013          Art                               Newell 31
33756      6  2013      Binders              Cardinal Hold-It CD Pocket
33757      6  2013       Labels                               Avery 487
33758     13  2013      Binders            Wilson Jones Binder, Economy

[13799 rows x 4 columns]

### total quantity of sales in decreasing order

In [31]:
sorted_year=df.groupby('year').sales.count().sort_values(ascending= False)
sorted_year

year
2014    17531
2013    13799
2012    10962
2011     8998
Name: sales, dtype: int64

### how much sales in africa region after 2013?

In [32]:
df[(df['year']>2013)& (df['region']=='Africa')]

order_id order_date  ship_date      segment              state  \
33765  AG-2014-2040 2014-01-01 2014-06-01     Consumer              Alger   
33771   CG-2014-920 2014-01-01 2014-02-01     Consumer           Kinshasa   
33776  AG-2014-2360 2014-01-01 2014-06-01     Consumer         Mostaganem   
33784  SU-2014-3390 2014-02-01 2014-06-01     Consumer            Red Sea   
33786  SU-2014-3390 2014-02-01 2014-06-01     Consumer            Red Sea   
...             ...        ...        ...          ...                ...   
51252  OD-2014-9490 2014-12-31 2015-05-01    Corporate  Central Equatoria   
51253  AO-2014-7730 2014-12-31 2015-04-01     Consumer             Luanda   
51262  TZ-2014-8350 2014-12-31 2015-03-01     Consumer      Dar Es Salaam   
51280  TZ-2014-8220 2014-12-31 2015-06-01  Home Office      Dar Es Salaam   
51286  MO-2014-2560 2014-12-31 2015-05-01     Consumer   Souss-Massa-Draâ   

                                country  market  region        product_id  \
33765                           Algeria  Africa  Africa  FUR-RUB-10003004   
33771  Democratic Republic of the Congo  Africa  Africa  OFF-KLE-10001644   
33776                           Algeria  Africa  Africa  OFF-CAM-10003933   
33784                             Sudan  Africa  Africa  TEC-SAM-10003493   
33786                             Sudan  Africa  Africa  OFF-ELD-10001477   
...                                 ...     ...     ...               ...   
51252                       South Sudan  Africa  Africa  TEC-CAN-10004291   
51253                            Angola  Africa  Africa  OFF-SAN-10001094   
51262                          Tanzania  Africa  Africa  OFF-FEL-10001792   
51280                          Tanzania  Africa  Africa  OFF-HAR-10001531   
51286                           Morocco  Africa  Africa  OFF-WIL-10001069   

              category sub_category  \
33765        Furniture  Furnishings   
33771  Office Supplies     Supplies   
33776  Office Supplies    Envelopes   
33784       Technology       Phones   
33786  Office Supplies      Storage   
...                ...          ...   
51252       Technology      Copiers   
51253  Office Supplies        Paper   
51262  Office Supplies      Storage   
51280  Office Supplies       Labels   
51286  Office Supplies      Binders   

                                         product_name sales  quantity  \
33765                       Rubbermaid Frame, Durable   107         1   
33771                         Kleencut Trimmer, Steel    41         1   
33776              Cameo Business Envelopes, Recycled    19         1   
33784                    Samsung Signal Booster, VoIP   547         4   
33786                     Eldon Lockers, Single Width   396         2   
...                                               ...   ...       ...   
51252                     Canon Wireless Fax, Digital   378         1   
51253                    SanDisk Memo Slips, 8.5 x 11   127         6   
51262                    Fellowes Folders, Wire Frame    23         1   
51280  Harbour Creations Removable Labels, Adjustable    50         6   
51286         Wilson Jones Hole Reinforcements, Clear     4         1   

       discount   profit  shipping_cost  year delivery_days  
33765       0.0   26.730          16.77  2014      151 days  
33771       0.0    6.510           6.72  2014       31 days  
33776       0.0    2.100           2.75  2014      151 days  
33784       0.0  185.880          41.63  2014      120 days  
33786       0.0  193.980          25.40  2014      120 days  
...         ...      ...            ...   ...           ...  
51252       0.0  166.440          11.71  2014      121 days  
51253       0.0   20.160          11.50  2014       91 days  
51262       0.1    4.053           6.11  2014       60 days  
51280       0.0    6.840           2.15  2014      152 days  
51286       0.0    0.420           0.49  2014      121 days  

[1624 rows x 19 columns]

### how many order placed in market of 'US' after 2013?

In [33]:
df[(df['year']>2013)&(df['market']=='US')]['order_id'].count()

3319

### number of the markets ?

In [34]:
df['market'].unique()

array(['Africa', 'APAC', 'EMEA', 'EU', 'Canada', 'LATAM', 'US'],
      dtype=object)

### Count of sales in different markets?

In [35]:
df.groupby('market').sales.count()

market
APAC      11002
Africa     4587
Canada      384
EMEA       5029
EU        10000
LATAM     10294
US         9994
Name: sales, dtype: int64

In [36]:
df.head(2)

order_id order_date  ship_date   segment            state    country  \
0   AG-2011-2040 2011-01-01 2011-06-01  Consumer      Constantine    Algeria   
1  IN-2011-47883 2011-01-01 2011-08-01  Consumer  New South Wales  Australia   

   market   region        product_id         category sub_category  \
0  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   

               product_name sales  quantity  discount   profit  shipping_cost  \
0       Tenex Lockers, Blue   408         2       0.0  106.140          35.46   
1  Acme Trimmer, High Speed   120         3       0.1   36.036           9.72   

   year delivery_days  
0  2011      151 days  
1  2011      212 days

### types of category?

In [37]:
df['category'].unique()

array(['Office Supplies', 'Furniture', 'Technology'], dtype=object)

### Count of office supplies ?

In [38]:
df[df['category']=='Office Supplies'].category.count()

31273

### number of furniture sold?

In [39]:
df[df['category']=='Furniture'].category.count()

9876

In [40]:
# number of Technology things sold
df[df['category']=='Technology'].category.count()


10141

In [41]:
df['sub_category'].unique()

array(['Storage', 'Supplies', 'Paper', 'Furnishings', 'Machines',
       'Appliances', 'Copiers', 'Chairs', 'Tables', 'Bookcases', 'Phones',
       'Accessories', 'Labels', 'Art', 'Envelopes', 'Fasteners',
       'Binders'], dtype=object)

In [42]:
# how much Bookcases were sold??
df.groupby('sub_category').sub_category.count()

sub_category
Accessories    3075
Appliances     1755
Art            4883
Binders        6152
Bookcases      2411
Chairs         3434
Copiers        2223
Envelopes      2435
Fasteners      2420
Furnishings    3170
Labels         2606
Machines       1486
Paper          3538
Phones         3357
Storage        5059
Supplies       2425
Tables          861
Name: sub_category, dtype: int64

### Three low sales sub_category product

In [43]:
df.groupby('sub_category').sub_category.count().sort_values(ascending = True).head(3)

sub_category
Tables         861
Machines      1486
Appliances    1755
Name: sub_category, dtype: int64

### Three top sales sub_category products?

In [44]:
df.groupby('sub_category').sub_category.count().sort_values(ascending = False).head(3)

sub_category
Binders    6152
Storage    5059
Art        4883
Name: sub_category, dtype: int64

### on "Supplies" sub_category product --> how many order placed??

In [45]:
df[df['sub_category']=='Supplies'].order_id.count()

2425

### Maximum sales How much?

In [46]:
df.sales.max()

'999'

### On which date , which state and which country sales was maximum??

In [47]:
df[df['sales']==df.sales.max()][['order_date','state','country','sales']]

order_date       state        country sales
5676  2011-09-27    Brittany         France   999
7602  2011-11-28       Texas  United States   999
8823  2011-12-28  California  United States   999
14710 2012-08-24       Texas  United States   999
40454 2014-06-28   Guatemala      Guatemala   999

In [48]:
df.head(2)

order_id order_date  ship_date   segment            state    country  \
0   AG-2011-2040 2011-01-01 2011-06-01  Consumer      Constantine    Algeria   
1  IN-2011-47883 2011-01-01 2011-08-01  Consumer  New South Wales  Australia   

   market   region        product_id         category sub_category  \
0  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   

               product_name sales  quantity  discount   profit  shipping_cost  \
0       Tenex Lockers, Blue   408         2       0.0  106.140          35.46   
1  Acme Trimmer, High Speed   120         3       0.1   36.036           9.72   

   year delivery_days  
0  2011      151 days  
1  2011      212 days

### How many type of segments and count of segments??

In [49]:
df.groupby('segment').segment.count()

segment
Consumer       26518
Corporate      15429
Home Office     9343
Name: segment, dtype: int64

### total profit earned by Store in dollers?

In [50]:
'$' + str(df['profit'].sum().round(2))

'$1469034.82'

### Total profit earned by store in 2012?

In [51]:
df[df['year']==2012]['profit'].sum().round(2)

307415.28

In [52]:
df.head()

order_id order_date  ship_date      segment            state  \
0     AG-2011-2040 2011-01-01 2011-06-01     Consumer      Constantine   
1    IN-2011-47883 2011-01-01 2011-08-01     Consumer  New South Wales   
2     HU-2011-1220 2011-01-01 2011-05-01     Consumer         Budapest   
3  IT-2011-3647632 2011-01-01 2011-05-01  Home Office        Stockholm   
4    IN-2011-47883 2011-01-01 2011-08-01     Consumer  New South Wales   

     country  market   region        product_id         category sub_category  \
0    Algeria  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1  Australia    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   
2    Hungary    EMEA     EMEA  OFF-TEN-10001585  Office Supplies      Storage   
3     Sweden      EU    North   OFF-PA-10001492  Office Supplies        Paper   
4  Australia    APAC  Oceania   FUR-FU-10003447        Furniture  Furnishings   

                  product_name sales  quantity  discount   profit  \
0          Tenex Lockers, Blue   408         2       0.0  106.140   
1     Acme Trimmer, High Speed   120         3       0.1   36.036   
2      Tenex Box, Single Width    66         4       0.0   29.640   
3  Enermax Note Cards, Premium    45         3       0.5  -26.055   
4   Eldon Light Bulb, Duo Pack   114         5       0.1   37.770   

   shipping_cost  year delivery_days  
0          35.46  2011      151 days  
1           9.72  2011      212 days  
2           8.17  2011      120 days  
3           4.82  2011      120 days  
4           4.70  2011      212 days

In [53]:
df['sales'] = df['sales'].str.replace(',', '').astype(int)


In [54]:
df.dtypes

order_id                  object
order_date        datetime64[ns]
ship_date         datetime64[ns]
segment                   object
state                     object
country                   object
market                    object
region                    object
product_id                object
category                  object
sub_category              object
product_name              object
sales                      int32
quantity                   int64
discount                 float64
profit                   float64
shipping_cost            float64
year                       int64
delivery_days    timedelta64[ns]
dtype: object

### Total sales of superstore?

In [55]:
df['sales'].sum()

12642905

### total sales by year ?

In [56]:
df.groupby('year').sales.sum().sort_values(ascending = False)

year
2014    4300041
2013    3405860
2012    2677493
2011    2259511
Name: sales, dtype: int32

### Avarage sales?

In [57]:
df['sales'].mean()

246.49844024176252

### which category product sold mostly in 2014 and how much?

In [58]:
df[df['year']==2014][['category']].groupby('category').category.count().sort_values(ascending= False).head(1)


category
Office Supplies    10729
Name: category, dtype: int64

In [59]:
df.head(2)

order_id order_date  ship_date   segment            state    country  \
0   AG-2011-2040 2011-01-01 2011-06-01  Consumer      Constantine    Algeria   
1  IN-2011-47883 2011-01-01 2011-08-01  Consumer  New South Wales  Australia   

   market   region        product_id         category sub_category  \
0  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   

               product_name  sales  quantity  discount   profit  \
0       Tenex Lockers, Blue    408         2       0.0  106.140   
1  Acme Trimmer, High Speed    120         3       0.1   36.036   

   shipping_cost  year delivery_days  
0          35.46  2011      151 days  
1           9.72  2011      212 days

In [60]:
df['sales'].describe().astype(int)

count    51290
mean       246
std        487
min          0
25%         31
50%         85
75%        251
max      22638
Name: sales, dtype: int32

#### How much % sales commited in 2011 from total sales?

In [61]:
((df[df['year']==2011]['sales'].sum())/(df['sales'].sum())*100).round(2) 

17.87

#### How much % sales commited in 2012 from total sales?

In [62]:
((df[df['year']==2012]['sales'].sum())/(df['sales'].sum())*100).round(2) 

21.18

#### How much % sales commited in 2013 from total sales???

In [63]:
((df[df['year']==2013]['sales'].sum())/(df['sales'].sum())*100).round(2) 

26.94

#### How much % sales commited in 2014 from total sales?

In [64]:
((df[df['year']==2014]['sales'].sum())/(df['sales'].sum())*100).round(2) 

34.01

In [65]:
df['total_sales']=df['sales']* df['quantity']

In [66]:
df.head()

order_id order_date  ship_date      segment            state  \
0     AG-2011-2040 2011-01-01 2011-06-01     Consumer      Constantine   
1    IN-2011-47883 2011-01-01 2011-08-01     Consumer  New South Wales   
2     HU-2011-1220 2011-01-01 2011-05-01     Consumer         Budapest   
3  IT-2011-3647632 2011-01-01 2011-05-01  Home Office        Stockholm   
4    IN-2011-47883 2011-01-01 2011-08-01     Consumer  New South Wales   

     country  market   region        product_id         category sub_category  \
0    Algeria  Africa   Africa  OFF-TEN-10000025  Office Supplies      Storage   
1  Australia    APAC  Oceania   OFF-SU-10000618  Office Supplies     Supplies   
2    Hungary    EMEA     EMEA  OFF-TEN-10001585  Office Supplies      Storage   
3     Sweden      EU    North   OFF-PA-10001492  Office Supplies        Paper   
4  Australia    APAC  Oceania   FUR-FU-10003447        Furniture  Furnishings   

                  product_name  sales  quantity  discount   profit  \
0          Tenex Lockers, Blue    408         2       0.0  106.140   
1     Acme Trimmer, High Speed    120         3       0.1   36.036   
2      Tenex Box, Single Width     66         4       0.0   29.640   
3  Enermax Note Cards, Premium     45         3       0.5  -26.055   
4   Eldon Light Bulb, Duo Pack    114         5       0.1   37.770   

   shipping_cost  year delivery_days  total_sales  
0          35.46  2011      151 days          816  
1           9.72  2011      212 days          360  
2           8.17  2011      120 days          264  
3           4.82  2011      120 days          135  
4           4.70  2011      212 days          570

### Total Sales with quantity?

In [67]:
df['total_sales'].sum()

61822900

### how much profit earned by superstore with respect to year?

In [68]:
df.groupby('year').profit.sum().sort_values(ascending= False)

year
2014    504165.97046
2013    408512.76018
2012    307415.27910
2011    248940.81154
Name: profit, dtype: float64

### Maximum Profit?

In [69]:
df['profit'].max()

8399.976

### On which category , segment, state ,sales and quantity where profit wasgiven maximum ?

In [70]:
df[df['profit']==df['profit'].max()][['segment','state','country','category','sales','quantity','year']]

segment    state        country    category  sales  quantity  year
29530  Corporate  Indiana  United States  Technology  17500         5  2013

##### to get better sales in Technology category products should be increase some discount

In [81]:
df[df['sales']==df['sales'].max()]

order_id order_date  ship_date      segment    state  \
1103  CA-2011-145317 2011-03-18 2011-03-23  Home Office  Florida   

            country market region       product_id    category sub_category  \
1103  United States     US  South  TEC-MA-10002412  Technology     Machines   

                                           product_name  sales  quantity  \
1103  Cisco TelePresence System EX90 Videoconferenci...  22638         6   

      discount     profit  shipping_cost  year delivery_days  total_sales  
1103       0.5 -1811.0784          24.29  2011        5 days       135828

#### On this order superstore got -1811.0784 loss in 2011.	